sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -653,6 +768,71 @@ def test_gaussian_mixture_fit():
             assert_allclose(ecov.error_norm(prec_pred[k]), 0, atol=0.15)
 
 
+def test_weighted_vs_repeated():
+    # a sample weight of N should yield the same result as an N-fold
+    # repetition of the sample
+    rng = np.random.RandomState(0)
+    n_samples = 200
+    rand_data = RandomData(rng, n_samples=n_samples)
+    n_repeated_samples = 80
+
+    n_features = rand_data.n_features
+    n_components = rand_data.n_components
+
+    for covar_type in COVARIANCE_TYPE:
+        X = rand_data.X[covar_type]
+
+        sample_weight = np.ones(n_samples)
+        sample_weight[n_repeated_samples:] += 2
+
+        X_second_component = X[n_repeated_samples:]
+        X_repeat = np.vstack((X, X_second_component, X_second_component))
+
+        g = GaussianMixture(n_components=n_components, n_init=20,
+                            reg_covar=0, random_state=rng,
+                            covariance_type=covar_type)
+
+        g1 = GaussianMixture(n_components=n_components, n_init=20,
+                             reg_covar=0, random_state=rng,
+                             covariance_type=covar_type)
+
+        g_weighted = g.fit(X, sample_weight=sample_weight)
+        g_repeated = g1.fit(X_repeat)
+
+        assert_allclose(np.sort(g_weighted.weights_),
+                        np.sort(g_repeated.weights_))
+
+        arg_idx1 = g_weighted.means_[:, 0].argsort()
+        arg_idx2 = g_repeated.means_[:, 0].argsort()
+        assert_allclose(g_weighted.means_[arg_idx1],
+                        g_repeated.means_[arg_idx2])
+
+        if covar_type == 'full':
+            prec_weighted = g_weighted.precisions_
+            prec_repeated = g_repeated.precisions_
+        elif covar_type == 'tied':
+            prec_weighted = np.array([g_weighted.precisions_] * n_components)
+            prec_repeated = np.array([g_repeated.precisions_] * n_components)
+        elif covar_type == 'spherical':
+            prec_weighted = np.array([np.eye(n_features) * c
+                                      for c in g_weighted.precisions_])
+            prec_repeated = np.array([np.eye(n_features) * c
+                                      for c in g_repeated.precisions_])
+        elif covar_type == 'diag':
+            prec_weighted = np.array([np.diag(d)
+                                      for d in g_weighted.precisions_])
+            prec_repeated = np.array([np.diag(d)
+                                      for d in g_repeated.precisions_])
+
+        arg_idx1 = np.trace(prec_weighted, axis1=1, axis2=2).argsort()
+        arg_idx2 = np.trace(prec_repeated, axis1=1, axis2=2).argsort()
+        for k, h in zip(arg_idx1, arg_idx2):
+            ecov = EmpiricalCovariance()
+            ecov.covariance_ = prec_weighted[h]
+            # the accuracy depends on the number of data and randomness, rng
+            assert_allclose(ecov.error_norm(prec_repeated[k]), 0, atol=0.001)
```

Comment:
are we not able to compare precisions directly once indexed by the appropriate `arg_idx`? I.e. can't we just `assert_allclose(g_weighted.precisions_[arg_idx1], g_repeated.precisions_[arg_idx2])` except where 	tied? I don't know this code well, admittedly.

Reply:
Here, I used a conservative approach as the Frobenius error norm was used in other tests, such as test_gaussian_mixture_fit. I changed for `assert_allclose(g_weighted.precisions_[arg_idx1], g_repeated.precisions_[arg_idx2])` and it failed for some tests.